ngraph/python/TUTORIAL.md

```diff
@@ -0,0 +1,216 @@
+
+```python
+>>> for node in function.get_ordered_ops():
+>>>     print('Node name: {:15} Friendly name: {:10} Op: {:10} {}'.format(
+>>>         node.name, node.get_friendly_name(), node.get_type_name(), node.shape))
+Node name: Parameter_1     Friendly name: b          Op: Parameter  Shape{2, 2}
+Node name: Parameter_0     Friendly name: x          Op: Parameter  Shape{2, 2}
+Node name: Add_2           Friendly name: Add_2      Op: Add        Shape{2, 2}
+Node name: Relu_3          Friendly name: y          Op: Relu       Shape{2, 2}
+Node name: Result_4        Friendly name: Result_4   Op: Result     Shape{2, 2}
+```
+
+##### Node relationships
+
+`Node`'s parents can be retrieved using the `inputs()` method, which returns `Input` objects. Each `Input` is a connection to an upstream `Output`, which is connected to a `Node`. The upstream node can be retrieved using `input.get_source_output().get_node()` methods.
+
+```python
+>>> for node_input in node.inputs():
+>>>     parent = node_input.get_source_output().get_node()
+>>>     print('{} node has parent {}'.format(node.get_friendly_name(), parent.get_friendly_name()))
+Add_2 node has parent x
+Add_2 node has parent b
+```
+
+By analogy, `Node`'s children can be retrieved using the `outputs()` method which returns a list of `Output` objects. By contrast to inputs, each output can be connected to multiple downstream nodes.
+
+```python
+>>> for node_output in node.outputs():
+>>>     children = [target.get_node() for target in node_output.get_target_inputs()]
+>>>     print('{} node has children {}'.format(node.get_friendly_name(), ','.join([c.get_friendly_name() for c in children])))
+>>> 
+Add_2 node has children y
+```
+
+##### Shapes
+
+Each `Input` and `Output` has an associated tensor shape, which describes what data it can accept or produce. Shapes can be dynamic, which means that they are only partially known. You can get information about this shape by calling the `get_partial_shape` method. If the returned `PartialShape` object's `is_dynamic` property returns `True`, then the shape is known partially. Otherwise you can convert the shape to a fully known static shape (list of integers).
+
+```python
+partial_shape = node_input.get_partial_shape()
+if partial_shape.is_dynamic:
+    print('{} node has a dynamic shape {}'.format(node.get_friendly_name(), partial_shape))
+else:
+    shape = list(partial_shape.to_shape())
+    print('{} node has a static shape {}'.format(node.get_friendly_name(), shape))
+```
+
+##### Node attributes
+
+Nodes may also have additional attributes. For example a `Softmax` node has an `axis` attribute. Each attribute may be retrieved using a dedicated method, for example `node.get_axis()`.
+
+```python
+>>> node = ng.softmax(data=[[1, 2], [3, 4]], axis=1)
+>>> node.get_axis()
```

Comment:
I think we need to add some words about set_XXX methods.

Reply:
Done.